grafana-plugin/src/containers/RemoteSelect/RemoteSelect.tsx

```diff
@@ -62,17 +63,22 @@ const RemoteSelect = inject('store')(
 
     const [options, setOptions] = useReducer(mergeOptions, []);
 
-    useEffect(() => {
-      makeRequest(href, {}).then((data) => {
-        setOptions(getOptions(data.results || data));
-      });
+    const loadOptionsCallback = useCallback((query?: string): void => {
+      (async () => {
+        try {
+          const data = await makeRequest(href, { params: { search: query } });
+          const options = getOptions(data.results || data);
+          setOptions(options);
+        } catch (e) {
+          if (axios.isAxiosError(e) && e.response.status === 403) {
+            // TODO: disable dropdown/update placeholder
```

Comment:
this is still a WIP. we probably need to allow passing a `requiredUserAction` prop to `RemoteSelect`. `requiredUserAction` should be of type `src/utils/authorization/index.ts:UserAction`.

In the case that the `RemoteSelect` component gets a 403 when trying to fetch/search options, we should disable the dropdown and update the placeholder (consider using the `src/utils/authorization/index.ts:generateMissingPermissionMessage` method for this).